### PR TITLE
Add go token type

### PR DESCRIPTION
**Pull request by louisruch**
### What does this PR do

- Adds TokenSecret and KeySecret go types to be able to redact sensitive data when stringer methods are invoked

### CODE REVIEW

**Comment by louisruch** [comment]
@mgaffney I also added a KeySecret since it was the other encrypted value

File: internal/credential/vault/secret.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+// MarshalJSON will redact the TokenSecret.
+func (s TokenSecret) MarshalJSON() ([]byte, error) {
+	return json.Marshal([]byte(redactedTokenSecret))
+}
+
+// KeySecret equals a Vault client certificate key.  This type provides a
+// wrapper so the secret isn't inadvertently leaked into a log or error.
+type KeySecret []byte
```
Comment (talanknight) [inline internal/credential/vault/secret.go:31]: This isn't used in the NewClientCertificate signature.  Should it be?

File: internal/credential/vault/secret.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+func (s TokenSecret) GoString() string {
+	return redactedTokenSecret
+}
+
+// MarshalJSON will redact the TokenSecret.
+func (s TokenSecret) MarshalJSON() ([]byte, error) {
+	return json.Marshal([]byte(redactedTokenSecret))
+}
+
+// KeySecret equals a Vault client certificate key.  This type provides a
```
Comment (mgaffney) [inline internal/credential/vault/secret.go:29]:
```suggestion
// KeySecret equals a Vault client certificate private key.  This type provides a
```

File: internal/credential/vault/secret.go (hunk 1 above)
Comment (mgaffney) [reply internal/credential/vault/secret.go:31]: +1

**Review by mgaffney — COMMENTED** [review]
I have the same question as @talanknight

**Pull request merged**